backend/src/web_server.py

```diff
@@ -62,37 +71,97 @@ def shutdown(self):
             (conn, address) = self.socket.accept()
             conn.settimeout(60)
-            threading.Thread(target=self.handleRequest, args=(conn, address)).start()
+            self.handleRequest(conn, address)
+
+        self.shutdown()
 
     def handleRequest(self, conn, address):
         """ Listens to the socket for incoming packets. When a incoming connection
             occurs a thread will be started to recieve and decode the packet.  """
-
+        # need to enable more threads processing and sending user packets. 
         data = str(conn.recv(PACKET_SIZE))
+
         # decode the packet into a dictionary type
-        # TODO: add client port info so we can send back data. 
+        # TODO: sending back data to front end ???
         strData = ast.literal_eval(str(data).strip("b\""))
-        # Add server to dict
+
         strData["server"] = SERV_HOST_NAME
-        command = strData["command"]
-        userCommands[command](strData)
+
+        if "user" in strData.keys():
+            # IF user field is specified send to the user process threads.
+            user = strData["user"]
+
+            if user not in self.userProcesses:
+                #IF no user process thread created initialize user process thread.
+                self.initializeUserThread(user)
+
+            threadContext = self.userProcesses[user]
+
+            # Add user command packet to thread work queue
+            threadContext["workQ"].put(strData)
+
+        else:
+            # NO user specified. Assume dumplog server command. 
+            threadContext = self.userProcesses["admin"]
+            threadContext["workQ"].put(strData)
+
+
+    def initializeUserThread(self, userId):
+        """ Every client has an asociated user thread. This thread will handle all of a specific user requests.
+            This function will initialize and start the user thread, including its queue, and user ports. """
+        
+        print("NEW USER! ", userId)
+        # Initialize the threadContext associated with the user thread. 
+        self.userProcesses[userId] = {
+            "userId" : userId,
+            "workQ"  : queue.Queue(),
+            "qLock"  : Lock(),
+        }
+
+        # Create and Start the user thread
+        cReqThread = Thread(target=self.handleClientRequest, args=[userId])
+        cReqThread.start()
+
+        return 
+
+    def handleClientRequest(self, userId):
+        """ This is the user thread associated with each user. Work is recieved from the client
+            and put in the associated user work thread queue. The thread will retrieve this work 
+            to execute on. """
+
+        print("Starting Thread Process for Client: ", userId)
+
+        threadContext = self.userProcesses[userId]
+
+        # TODO: allow thread closing here through threadContext. 
+        while self.serverRunning:
+            # Wait for next work Q item.
+            userReq = threadContext["workQ"].get()
```

Comment:
Retrieve user command dictionary from the thread specific queue in the thread context.